XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs

```diff
@@ -212,24 +208,156 @@ public override void OnPostRaycast()
         public override void OnInputChanged(InputEventData<Vector2> eventData)
         {
             // Don't process input if we've got an active teleport request in progress.
-            if (IsTeleportRequestActive || !IsTeleportSystemEnabled) { return; }
+            if (eventData.used || IsTeleportRequestActive || !IsTeleportSystemEnabled)
+            {
+                return;
+            }
 
             if (eventData.SourceId == InputSourceParent.SourceId &&
                 eventData.Handedness == Handedness &&
-                eventData.MixedRealityInputAction == teleportAction)
+                eventData.MixedRealityInputAction == MixedRealityToolkit.TeleportSystem.TeleportAction)
+            {
+                eventData.Use();
+                ProcessDualAxisTeleportInput(eventData);
+            }
+        }
+
+        #endregion IMixedRealityInputHandler Implementation
+
+        #region IMixedRealityTeleportHandler Implementation
+
+        /// <inheritdoc />
+        public override void OnTeleportRequest(TeleportEventData eventData)
+        {
+            // Only turn off the pointer if we're not the one sending the request
+            if (eventData.Pointer.PointerId == PointerId)
+            {
+                IsTeleportRequestActive = false;
+            }
+            else
+            {
+                IsTeleportRequestActive = true;
+                BaseCursor?.SetVisibility(false);
+            }
+        }
+
+        /// <inheritdoc />
+        public override void OnTeleportCompleted(TeleportEventData eventData)
+        {
+            IsTeleportRequestActive = false;
+            BaseCursor?.SetVisibility(false);
+        }
+
+        /// <inheritdoc />
+        public override void OnTeleportCanceled(TeleportEventData eventData)
+        {
+            IsTeleportRequestActive = false;
+            BaseCursor?.SetVisibility(false);
+        }
+
+        #endregion IMixedRealityTeleportHandler Implementation
+
+        private void ProcessDigitalTeleportInput(bool isPressed)
+        {
+            currentDigitalInputState = isPressed;
+
+            if (currentDigitalInputState && !teleportEnabled)
+            {
+                teleportEnabled = true;
+                MixedRealityToolkit.TeleportSystem?.RaiseTeleportRequest(this, TeleportHotSpot);
+            }
+            else if (!currentDigitalInputState)
             {
-                currentInputPosition = eventData.InputData;
+                var canTeleport = false;
+
+                if (teleportEnabled &&
+                TeleportValidationResult == TeleportValidationResult.Valid ||
+                TeleportValidationResult == TeleportValidationResult.HotSpot)
+                {
+                    canTeleport = true;
+                }
+
+                if (canTeleport)
+                {
+                    teleportEnabled = false;
+
+                    if (TeleportValidationResult == TeleportValidationResult.Valid ||
```

Comment:
Could this code section be broken out into its own function as it's repeated quite a few times throughout the script?

Reply:
Probably can but wasn't as trivial as I thought and broke dual axis in some cases. I reverted it. Let's keep it like this for now. I have some more work for the TeleportPointer on my list where I will tackle this.